feat(moontest/init.lua): changed dofile to one function

ApolloX939 opened the pull request:
This change now makes it very easy to change the file structure because there is one spot

> Rather than having 18 lines to change make it so it's just 1 line. (DRY)

Commit:
feat(moontest/init.lua): changed dofile to one function

This change now makes it very easy to change the file structure because there is one spot

ApolloX939 commented on mods/moontest/init.lua:
```diff
@@ -24,24 +24,30 @@ minetest.settings:set_bool("smooth_lighting", true)
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "tutorial.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "shop_formspec.lua")
+-- Executes source files, just give .lua filename from src directory
+local do_src = function(filename)
```
I make this function local so it doesn't conflict with any future mods that could be added to this game.

> This allows for just passing the filename and it executing that file from source directory. (Making any changes to directory structure easy to change, since it's in one spot)

Droog71 replied:
I like how the file name is passed to the function so the long form is only used once.

Not knowing about this pull request at the time, I moved all the do_file stuff into a file called do_file.lua and now just call do_file on that file and it calls it for the rest.
(https://github.com/Droog71/moontest/blob/main/mods/moontest/src/do_file.lua)

Maybe the do_src function could be moved to do_file.lua. 

I thought about doing something like this:

```
local path = minetest.get_modpath("moontest") .. "/src/"
local pfile = io.popen('ls -a "'..path..'"')
for filename in pfile:lines() do
    do_file(filename)
end
pfile:close()
```

Unfortunately, this would require adding the mod to the trusted mods list.

ApolloX939 replied:
Yup, just added it instead it's now do_file rather than do_src. (in do_file.lua)

Commit:
style(moontest/init.lua): removed excess white-space after hud message

I was going to reduce respawn costs from 2000 to 1000 (50% off) but after playing a while with a fully automated base it doesn't really matter then.

Commit:
fix(moontest/src/aliens.lua): fixed aliens removing themselves when they enter wires

Aliens will now check if it's a mesecons node, if so then they are permitted to go thru it.

Fixes issue #7

Commit:
fix(aliens.lua): reversed change to copy original

It was found that aliens checking was no longer needed because of a bug in a previous version, thus was removed in the original repo.

BREAKING CHANGE: changes the repo into a somewhat mergable state from original repo (to update from 1.0.3 to 1.0.4)

Commit:
Merge branch 'Droog71:main' into main

Commit:
refactor(do_file.lua): made function to simplify dofile calls

This change makes it so there is one place to change all the files, thus making it more managable and more DRY. (Don't know DRY, google `programming dry principle`)

This brings this fork to 1.0.4 (up-to-date)

ApolloX939 commented:
The later changes were on merging my fork so it's current, so this PR is only to change do_file.lua in src directory.

Droog71 commented:
Thanks! Looks good.
